client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -89,9 +91,19 @@ tie.factory('PythonCodeRunnerService', [
         }
 
         // The run was successful.
-        return CodeEvalResultObjectFactory.create(
+        var codeEvalResult = CodeEvalResultObjectFactory.create(
           code, outputLines.join('\n'), observedOutputs,
           buggyOutputTestResults, performanceTestResults, null, null);
+        var question = CurrentQuestionService.getCurrentQuestion();
+        if (!question) {
+          return codeEvalResult;
+        }
+        var tasks = question.getTasks();
+        var testToDisplay = codeEvalResult.getIndexOfFirstFailedTest(tasks);
```

Comment:
A few issues here:

- This doesn't seem to be relevant to the purpose of this service, which is to run the code and return the result of doing so.
- It's not a good idea to redefine outputLines in the middle of a function. It makes it hard to tell what the variable represents since it refers to different things at different places.
- I'm not sure why outputLines is being modified in the first place. Where is it being used after this?

From a conceptual design perspective, probably better to just store the output-per-test-case in the codeEvalResult and refer to it (or the relevant part of it) later, when needed.

Reply:
Done. Thanks for the good suggestion as well!